tavern_server/src/db_tables.sql

```diff
@@ -0,0 +1,676 @@
+        char_id         	UUID,
+        race_id         	UUID,
+        deity_id        	UUID,
+
+        name            	TEXT,
+        age             	INT,
+        gender          	Gender,
+        alignment       	Alignment,
+        backstory       	TEXT,
+        height          	INT,
+        weight          	INT,
+        size            	Size,
+
+        strength        	INT,
+        dexterity       	INT,
+        constitution    	INT,
+        intelligence    	INT,
+        wisdom          	INT,
+        charisma        	INT,
+
+        max_hp          	INT,
+        damage          	INT,
+        nonlethal       	INT,
+
+        copper          	INT,
+        silver          	INT,
+        gold            	INT,
+        platinum        	INT
+);
+
+--Race tables
+CREATE TABLE Races (
+	race_id 		UUID,
+
+	name			TEXT,
+	move_speed		INT,
+	size			Size,
+	languages		TEXT,
+	race_type		RaceType
+);
+
+--Class tables
+CREATE TABLE Classes (
+	class_id		UUID,
+
+	name			TEXT,
+	hit_die			TEXT,
+	starting_wealth		TEXT,
+	bab_per_level		REAL,
+	skills_per_level	INT,
+	skills_attr		Attribute
+);
+
+CREATE TABLE Subclasses (
+	subclass_id		UUID,
+	class_id		UUID,
+
+	caster_type		CasterType,
+	casting_attr		Attribute
+);
+
+CREATE TABLE characterSubclasses (
+	char_id			UUID,
+	subclass_id		UUID,
+
+	levels_taken		INT,
+	hp_taken		INT,
+	skills_taken		INT
+);
+
+--Feats tables
+CREATE TABLE Feats (
+	feat_id			UUID,
+	req_id			UUID,
+
+	short_description	TEXT,
+	long_description	TEXT
+);
+
+CREATE TABLE SkillReqUnits (
+	req_id			UUID,
+	skill_unit_id		UUID
+);
+
+CREATE TABLE SkillFeatUnits (
+	skill_unit_id		UUID,
+	req_skill		Skill,
+	ranks			INT
+);
+
+CREATE TABLE AttributeReqUnits (
+	req_id			UUID,
+	attr_unit_id		UUID
+);
+
+CREATE TABLE AttributeFeatUnits (
+	attr_unit_id		UUID,
+	req_attr		Attribute,
+	score			INT
+);
+
+--this table creates a many to many realationship internally between feats.
+--any given feat can require a number of other feats to be taken,
+--and a given feat can be required by any number of other feats.
+--this table accomplishes that.
+CREATE TABLE RequiredFeats (
+	req_id			UUID,
+	feat__id		UUID
+);
+
+CREATE TABLE CharacterFeats (
+	char_id			UUID,
+	feat_id			UUID
+);
+
+CREATE TABLE RacialFeats (
+	race_id			UUID,
+	feat_id			UUID
+);
+
+CREATE TABLE ClassFeats (
+	class_id		UUID,
+	feat_id			UUID
+);
+
+--Features tables
+CREATE TABLE Features (
+	feature_id		UUID,
+	description		TEXT
+);
+
+CREATE TABLE ClassFeatures (
+	class_id		UUID,
+	feature_id		UUID,
+	is_default		BOOLEAN
+);
+
+CREATE TABLE CharacterFeatures (
+	char_id			UUID,
+	feature_id		UUID
+);
+
+CREATE TABLE SubclassFeatures (
+	subclass_id		UUID,
+	feature_id		UUID
+);
+
+--class proficiencies tables
+CREATE TABLE ClassProficientSingleWeapon (
+	class_id		UUID,
+	item_id			UUID
+);
+
+CREATE TABLE ClassProficientWeaponClass (
+	class_id		UUID,
+	weapon_class		WeaponClass
+);
+
+CREATE TABLE ClassProficientWeaponAllBut (
```

Comment:
I think we discussed this before: perhaps it would be better to indicate weapons that the class specifically is not proficient in? Then that could be combined with the weapon type proficiencies to create the same effect while being more flexible

Reply:
The suite of tables here was meant to accomplish this, but without arrays. Now that I know that PostgreSQL supports columns with array types, this has been cleaned up as you suggested. I feel like it's more intuitive for the table to contain the weapons a class *is* proficient in, though. Implemented all the same.